From c8b1e9c45a5bc5374224fedea1406a1975b5c304 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Wed, 5 Jul 2017 15:59:59 +0300
Subject: [PATCH 2077/5242] MLK-15317-3: ASoC: fsl_mqs: Add support for MQS
 dummy codec in imx8qxp

commit  91a385f30d76416768fc725863632752c0a9aefc from
https://source.codeaurora.org/external/imx/linux-imx.git

MQS on i.mx8 QXP uses the same register as i.mx8 QM
so we only need to add compatible string here.

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/sound/fsl,mqs.txt          |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi     |   10 ++++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/sound/fsl,mqs.txt b/Documentation/devicetree/bindings/sound/fsl,mqs.txt
index 092083d..4e81e90 100644
--- a/Documentation/devicetree/bindings/sound/fsl,mqs.txt
+++ b/Documentation/devicetree/bindings/sound/fsl,mqs.txt
@@ -3,7 +3,7 @@ fsl,mqs audio CODEC
 Required properties:
 
   - compatible : must contain one of "fsl,imx6sx-mqs", "fsl,codec-mqs"
-		"fsl,imx8qm-mqs".
+		"fsl,imx8qm-mqs", "fsl,imx8qxp-mqs".
 
   - clocks : a list of phandles + clock-specifiers, one for each entry in
     clock-names
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 52a9d16..0def446 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -1395,6 +1395,16 @@
 		status = "disabled";
 	};
 
+	mqs: mqs@59850000 {
+		compatible = "fsl,imx8qm-mqs";
+		reg = <0x0 0x59850000 0x0 0x10000>;
+		clocks = <&clk IMX8QXP_AUD_MQS_IPG>,
+			<&clk IMX8QXP_AUD_MQS_HMCLK>;
+		clock-names = "core", "mclk";
+		power-domains = <&pd_mqs0>;
+		status = "disabled";
+	};
+
 	usdhc1: usdhc@5b010000 {
 		compatible = "fsl,imx8qm-usdhc", "fsl,imx6sl-usdhc";
 		interrupt-parent = <&gic>;
-- 
1.7.9.5

